Makefile

```diff
@@ -2,7 +2,19 @@ SHELL := /bin/sh
 CARGO = cargo
 TESTS_DIR := tests
 TEST_RESULTS_DIR ?= workspace/test-results
-PYTEST_ARGS ?= $(if $(SKIP_SENTRY),-m "not sentry") $(if $(TEST_STUB),,-m "not stub") # Stub tests do not work in CI
+# Markers are "clever", and work by including multiple markers with " and ".
+# This uses makefile magic to construct the set of items.
+# technically `$(eval )` would work here, but I've not had it work consistently.
+# The following is taken from [the makefile manual](https://www.gnu.org/software/make/manual/make.html#Syntax-of-Functions)
+NOOP :=
+SPACE :=$(NOOP) $(NOOP)
+PYTEST_AND :=" and "
+# specify the tests to NOT run
+# NOTE: passing `-m not` to pytest will cause an error be sure at least one of these is set.
+NEG_PYTEST_ARGS ?= $(if $(SKIP_SENTRY),sentry) $(if $(TEST_STUB),,stub) $(if $(TEST_RELIABILITY),,reliable_report) # Stub tests do not work in CI
+# Now compose the pytest args by gluing the above items together
+NEG_PYTEST_ARGS := "not $(subst $(SPACE), and not ,$(strip $(NEG_PYTEST_ARGS)))"
+PYTEST_ARGS := -m $(NEG_PYTEST_ARGS)
```

Comment:
I hadn't realized that multiple -m flags are not additive TIL. My thought to overcome this lookes like this, it's less efficient but imo easier to understand. 

```
PYTEST_MARKERS := $(if $(SKIP_SENTRY),not sentry,)
PYTEST_MARKERS += $(if $(TEST_STUB),,$(if $(strip $(PYTEST_MARKERS)),and ,)not stub)
PYTEST_MARKERS += $(if $(TEST_RELIABILITY),,$(if $(strip $(PYTEST_MARKERS)),and ,)not reliable_report)
PYTEST_ARGS ?= $(if $(strip $(PYTEST_MARKERS)),-m "$(strip $(PYTEST_MARKERS))",)
```

That said, perhaps controlling markers with environment variables is overall a poor choice and we should perhaps stick to setting their values in the configuration, as @taddes suggested. We could add the line `addopts = -m "not stub and not reliable_report"` under `markers` in the [pyproject.toml](https://github.com/mozilla-services/autopush-rs/blob/master/tests/pyproject.toml#L100-L103)


**Also some reminders:**

nit: please add the reliable_report marker to the list in [pyproject.toml](https://github.com/mozilla-services/autopush-rs/blob/master/tests/pyproject.toml#L100-L103)
<img width="959" alt="image" src="https://github.com/user-attachments/assets/6b4ae795-196e-48c4-b657-1ee09f58e995">

nit: please include the reliable_report marker in the [testing.md documentation](https://github.com/mozilla-services/autopush-rs/blob/master/docs/src/testing.md?plain=1#L91-L94)

Reply:
I'm fine with less efficient / more readable. I'm not sure adding them to `pyproject.toml` is a good idea, for a couple of reasons. I would prefer if test configuration was done at one location, so we don't have to chase around if we want to change things later. It would be nice to have some sort of "combination" function like `standard = "not stub and not reliable_report and not sentry"` but I can see that getting equally complex in the future and I'd like to not make test configurations Turing complete.

Reply:
Related: https://github.com/pytest-dev/pytest/issues/7229